moveit_kinematics/ikfast_kinematics_plugin/templates/ikfast61_moveit_plugin_template.cpp

```diff
@@ -1060,13 +1093,41 @@ bool IKFastKinematicsPlugin::getPositionIK(const geometry_msgs::Pose& ik_pose, c
     ROS_DEBUG_STREAM_NAMED("ikfast", "No IK solution");
   }
 
+  // Sort the solutions under limits and find the one that is closest to ik_seed_state
```

Comment:
On second thought, the implementation might be correct, and `swap_flag` was only intended as an optimization... anyway, why waste brain cycles, let's just use std::sort. :)